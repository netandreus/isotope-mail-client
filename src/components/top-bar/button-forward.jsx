import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import TopBarButton from './top-bar-button';

export const ButtonForward = ({t, outboxEmpty, forwardMessage}) => (
  outboxEmpty && (
    <span isotip={t('topBar.forward')} isotip-position='bottom' isotip-size='small'>
      <TopBarButton onClick={forwardMessage}>forward</TopBarButton>
    </span>)
);

ButtonForward.propTypes = {
  outboxEmpty: PropTypes.bool.isRequired,
  forwardMessage: PropTypes.func.isRequired
};

export default translate()(ButtonForward);
