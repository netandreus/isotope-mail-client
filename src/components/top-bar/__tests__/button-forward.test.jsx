import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ButtonForward} from '../button-forward';

describe('ButtonForward component test suite', () => {
  test('Snapshot render, outboxEmpty, should render button', () => {
    // Given
    const props = {t: jest.fn(arg => arg), outboxEmpty: true, forwardMessage: jest.fn()};
    // When
    const button = shallow(<ButtonForward {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
  test('Snapshot render, not outboxEmpty, should NOT render button', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: false, forwardMessage: jest.fn()};
    // When
    const button = shallow(<ButtonForward {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
});
