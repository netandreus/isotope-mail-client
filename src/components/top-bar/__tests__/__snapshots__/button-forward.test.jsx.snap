// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonForward component test suite Snapshot render, not outboxEmpty, should NOT render button 1`] = `""`;

exports[`ButtonForward component test suite Snapshot render, outboxEmpty, should render button 1`] = `
<span
  isotip="topBar.forward"
  isotip-position="bottom"
  isotip-size="small"
>
  <TopBarButton
    className=""
    onClick={[MockFunction]}
  >
    forward
  </TopBarButton>
</span>
`;
